src/ledgers/utils.ts

```diff
@@ -433,28 +435,44 @@ export const unknownMessage = {
   displayName: 'Unknown'
 }
 
-export const signingClient = (async () => {
-  try {
+const switchLedgerType = async (
+  ledgerType: string
+): Promise<OfflineSigner | undefined> => {
+  let client
+  switch (ledgerType) {
+    case CosmosNetworkConfig.KEPLR_LEDGER:
+      client = await window.getOfflineSigner(import.meta.env.VITE_APP_CHAIN_ID)
+      return client
+    case CosmosNetworkConfig.COSMOSTATION_LEDGER:
+      client = await cosmostationSigner(import.meta.env.VITE_APP_CHAIN_ID)
+      return client
+    default:
+      return undefined
+  }
+}
+
```

Comment:
switchLedgerType() again appears here as a locally defined function, but it does different job compared to the ones in App.tsx and ConnectWallet.tsx, please review namings.

Reply:
Done